### DISCUSSION TITLE
Abnormally high phonon linewidth (or small phonon lifetime) for larger supercell

### DISCUSSION THREAD

**Discussion by ballal09** [ttadano/alamode#142] [General]
Dear Dr. Tadano,

I hope you are doing well.  I am trying to calculate the phonon lifetime for my system using 3rd order force constant generated by the ALAMODE.  First, I have used a smaller supercell (1 1 4 supercell) and found the results to be reasonable to some extent (for complex crystal ) except the acoustic branches. My 1 1 4 supercell contains some negative phonon branches for the low frequency modes (acoustic and optical) which go away when the supercell size is increased to 2 2 4. Having observed that, I have performed the phonon lifetime calculation using 2 2 4 supercell and found abnormally high value for the phonon linewidth which seems unphysical. I have redone the calculation but found the same result. I am expecting phonon lifetime to be more accurate with the larger supercell, however I am observing the unphysical behavior. Probably, there is something wrong in my calculation. FYI, fitting error for  harmonic and cubic fcs are satisfactory for both  supercell. I also like to mention that I have used the same cutoff for cubic fc calculation for both supercell. Again, I have used the right conversion factor while converting phonon linewidth (1/cm) to lifetime (ps) as you mentioned in a previous post. I would highly  appreciate your valuable feedback. I have attached  phonon lifetime result for both supercell, a snapshot of the result file and the RTA input script. Please let me know if you need further info regarding the calculation. Thank you very much for your time.

![tau_S114](https://github.com/ttadano/alamode/assets/60304718/e953a23a-a9bc-450b-bb09-f17c1d8907d3)
![tau_S224](https://github.com/ttadano/alamode/assets/60304718/87f9166a-5641-440b-9328-4daf93e86afa)
![result_S114](https://github.com/ttadano/alamode/assets/60304718/13cbba47-8e32-477f-997c-1dc9f83f9b5c)
![result_S224](https://github.com/ttadano/alamode/assets/60304718/6905354b-5f9d-4d9a-8003-2158c3690882)
![RTA_smaple](https://github.com/ttadano/alamode/assets/60304718/38ab7585-cd11-4583-9e7a-1ed275717bb6)

**Comment by ttadano** [comment]
In general, lifetimes (inverse scattering rates) and thermal conductivities calculated using phonons with imaginary branches are not reliable. In ANPHON, these imaginary branches are neglected when computing the scattering rates and modal contribution to the thermal conductivity, so if there are many imaginary branches, they will surely influence the values of lifetimes and thermal conductivity.

Anyway, in your case of the 2x2x4 supercell, the values of the scattering rate are indeed (unphysically) large. Perhaps, some of the anharmonic force constants can be very large due to the failure of fitting. Please check if force constants in `PREFIX.fcs` are reasonable in their absolute values.

**Comment by ballal09** [comment]
Dear Dr. Tadano

Thank you very much for your quick response. I do see some large value (in the order of e+02) in the the cubic.fcs file for the 2 2 4 supercell. I have attached a snapshot of the  cubic.fcs file for 2 2 4 supercell for your reference. Please let me know if you think having a look at the fcs file would be helpful (I can send that). My 2 2 4 supercell  phonon dispersion does not have negative modes, that's why I am expecting to obtain better result (phonon lifetime) with larger supercell. But it is still not clear to me why with the larger supercell I am having such larger order of magnitude  mismatch in phonon lifetime. FYI, My fitting errors for cubic fcs are 1.5% for 1 1 4 supercell and  2.6% for 2 2 4 supercell. I would highly appreciate you valuable feed back on this. Thank you very much for your time.

Best
Ballal
![S224_cubic_fcs_snap](https://github.com/ttadano/alamode/assets/60304718/1da12acc-2b2c-49a1-a3b8-2ba7d4e6fc84)

**Reply by ttadano** [reply]
Yes, these third-order force constants are unusually large. 

Please test if the following resolves the problem:
* Subtract the residual forces from `DFSET` using `--offset` option of extract.py. About the `--offset` option, please have a look at https://alamode.readthedocs.io/en/latest/faq.html.
* Use larger magnitude of displacements for the third-order term, say 0.03–0.05 Angstrom.

**Comment by ballal09** [comment]
Thank you very much for the quick response, I will test the following cases.

**Comment by ballal09** [comment]
Dear Dr. Tadano

I have implemented the suggestions you made for the 2 2 4 supercell.  While using the offset option  both the harmonic and cubic fc fitting error reduced further.  My  harmonic fitting error is 1.67% and cubic one is 1.42%  which are very much acceptable and also there was no imaginary branches in the harmonic phonon dispersion. My displacement magnitude for the harmonic case is 0.01 A, but increased the cubic displacement to 0.08 A. However, the magnitude of the phonon linewidth (1/cm) is still unphysically high. I have attached an image of that.  One trend that I noticed while playing with the cubic displacement magnitude is that increasing cubic displacement magnitude reduces the linewidth (1/cm) by two-three order of magnitude but they are still unphysically high. I have also done the same calculation for couple of other supercell like 2 2 1, 2 2 3 but no improvement. I am curious what is  going wrong when increasing the supercell size from 1 1 4 to 2 2 4, although the fitting errors are pretty acceptable. I would highly appreciate any further insights. Thank you for your time.

![result_S224_0 08A](https://github.com/ttadano/alamode/assets/60304718/7e26e1e7-2a1c-4c22-b4f8-012f0d9a4e57)

**Reply by ttadano** [reply]
Hi,

For the `&cell` field of **ANPHON**, you need to give the lattice parameters of the **primitive cell** (not supercell). Can you check if you did so correctly?

**Comment by ballal09** [comment]
Hi Dr. Tadano

I have checked that carefully. For both 1 1 4 and 2 2 4 supercell, I have used the primitive cell lattice parameter in the &cell field of anphon and  supercell lattice parameter in the &cell field of alm. FYI, for the 2 2 4 supercell I have used sparse=1 in the optimize section due to large data set, but this should not be an issue. Please let me know if you need any other info to get better insight. Thank you very much for your response.

**Reply by ttadano** [reply]
I think `SPARSE = 1` will not influence accuracy. Another point I would suggest is to see the dependency on the cutoff radius for the third-order terms. Do you still see the problem when you use a smaller cutoff radius?

**Comment by ballal09** [comment]
Thank you for the  quick response. I will give a try with smaller cutoff and let you know what happens.

**Comment by ballal09** [comment]
Dear Dr. Tadano
It worked when I reduced my cut off distance for the larger supercell. Obtained phonon lifetime is now physically meaningful  and makes sense  when I compare it with similar structure with the literature. I am just curious what went wrong with the relatively large cutoff distance with the larger supercell although it worked perfectly with the smaller supercell.  Again thank you very much for your insightful suggestions.
